File: Fundamentals_of_Software_Design/1_Basic_skills/Цельсий и Фаренгейт/main.cpp
#include <iostream>
#include <iomanip> // Для настройки вывода с фиксированным количеством знаков после запятой

double celsius_to_fahrenheit(double celsius) {
    return celsius * 9.0 / 5.0 + 32.0;
}

double fahrenheit_to_celsius(double fahrenheit) {
    return (fahrenheit - 32.0) * 5.0 / 9.0;
}

int main() {
    int choice;
    double temperature, converted;

    std::cout << "Выберите режим перевода:\n";
    std::cout << "1. Цельсий в Фаренгейт\n";
    std::cout << "2. Фаренгейт в Цельсий\n";
    std::cout << "Введите номер режима (1 или 2): ";
    std::cin >> choice;

    if (choice == 1) {
        std::cout << "Введите температуру в градусах Цельсия: ";
        std::cin >> temperature;
        converted = celsius_to_fahrenheit(temperature);
        std::cout << std::fixed << std::setprecision(2);
        std::cout << temperature << "°C = " << converted << "°F\n";
    } else if (choice == 2) {
        std::cout << "Введите температуру в градусах Фаренгейта: ";
        std::cin >> temperature;
        converted = fahrenheit_to_celsius(temperature);
        std::cout << std::fixed << std::setprecision(2);
        std::cout << temperature << "°F = " << converted << "°C\n";
    } else {
        std::cout << "Некорректный выбор режима.\n";
    }

    return 0;
}

File: Fundamentals_of_Software_Design/1_Basic_skills/Цельсий и Фаренгейт/main.py
def celsius_to_fahrenheit(celsius):
    return celsius * 9 / 5 + 32

def fahrenheit_to_celsius(fahrenheit):
    return (fahrenheit - 32) * 5 / 9

def main():
    print("Выберите режим перевода:")
    print("1. Цельсий в Фаренгейт")
    print("2. Фаренгейт в Цельсий")
    
    choice = input("Введите номер режима (1 или 2): ")
    
    if choice == "1":
        celsius = float(input("Введите температуру в градусах Цельсия: "))
        fahrenheit = celsius_to_fahrenheit(celsius)
        print(f"{celsius:.2f}°C = {fahrenheit:.2f}°F")
    elif choice == "2":
        fahrenheit = float(input("Введите температуру в градусах Фаренгейта: "))
        celsius = fahrenheit_to_celsius(fahrenheit)
        print(f"{fahrenheit:.2f}°F = {celsius:.2f}°C")
    else:
        print("Некорректный выбор режима.")

if __name__ == "__main__":
    main()
